### ISSUE TITLE
clang mixes up `namespace` with `class`  in inheritance and `using namespace`

### ISSUE THREAD

**Issue by Rush10233** [llvm/llvm-project#101189]
The following code is rejected by clang by ambiguous reference to  `X`:

```c++
namespace X{
    struct X{

    };
}
using namespace X;
struct Y:X{};
//OK
//struct Y:X::X{};
```
The diagnostic is quite unreasonable:

```c++
<source>:7:10: error: reference to 'X' is ambiguous
    7 | struct Y:X{};
      |          ^
<source>:1:11: note: candidate found by name lookup is 'X'
    1 | namespace X{
      |           ^
<source>:2:12: note: candidate found by name lookup is 'X::X'
    2 |     struct X{
      |            ^
1 error generated.
```

There should not be any ambiguity because namespace `X` definitely cannot be treated as a candidate base type of struct `Y`, and the reference to `X` in `Y:X`  is clearly `X::X`.

As a comparison, directly using `X::X` without `using namespace` does not cause any error.

GCC and MSVC both accept the code.

[https://godbolt.org/z/xWxYMbMYs](https://godbolt.org/z/xWxYMbMYs)

**Comment by AaronBallman** [comment]
The base class lookup is type-only per https://eel.is/c++draft/class.derived#general-2.sentence-4 and that should not find namespaces per https://eel.is/c++draft/basic.lookup#general-4.sentence-4

**Comment by shafik** [comment]
This goes all the way back to clang-3: https://godbolt.org/z/Kn3zPxqon
